dominion-pax-exam-driver/src/main/java/org/codice/dominion/pax/exam/internal/DominionConfigurationFactory.java

```diff
@@ -0,0 +1,424 @@
+      throw new ConditionException(
+          "failed to evaluate condition from "
+              + annotation.annotationType().getSimpleName()
+              + " extension for "
+              + testClass.getName()
+              + " and "
+              + annotation,
+          t);
+    }
+  }
+
+  /** Wraps all annotations defining options into one single composite option. */
+  private class AnnotationOptions implements CompositeOption {
+    private final List<ExtensionOption> options;
+
+    AnnotationOptions(Stream<AnnotationEntry<Option.Annotation>> annotations) {
+      this.options =
+          annotations
+              .map(ExtensionOption::new)
+              .flatMap(ExtensionOption::extensions)
+              .collect(Collectors.toList());
+    }
+
+    private boolean isForSystemOS(KarafDistributionBaseConfigurationOption option) {
+      if (option instanceof KarafDistributionKitConfigurationOption) {
+        final KarafDistributionKitConfigurationOption kit =
+            (KarafDistributionKitConfigurationOption) option;
+
+        if (SystemUtils.IS_OS_WINDOWS) {
+          return Platform.WINDOWS.equals(kit.getPlatform());
+        }
+        return !Platform.WINDOWS.equals(kit.getPlatform());
+      }
+      return true;
+    }
+
+    public KarafDistributionBaseConfigurationOption getDistribution() {
+      final KarafDistributionBaseConfigurationOption[] distros =
+          options
+              .stream()
+              .map(ExtensionOption::getOptions)
+              .flatMap(Stream::of)
+              .filter(KarafDistributionBaseConfigurationOption.class::isInstance)
+              .map(KarafDistributionBaseConfigurationOption.class::cast)
+              .filter(this::isForSystemOS)
+              .toArray(KarafDistributionBaseConfigurationOption[]::new);
+
+      if (distros.length > 1) { // we don't support more than one yet
+        throw new OptionException(
+            "too many Karaf distributions configured for the same OS; only one is currently supported");
+      } else if (distros.length == 0) {
+        throw new OptionException(
+            "no configured Karaf distributions; annotate "
+                + testClass.getName()
+                + " with @"
+                + KarafOptions.DistributionConfiguration.class.getSimpleName());
+      }
+      return distros[0];
+    }
+
+    @Override
+    public org.ops4j.pax.exam.Option[] getOptions() {
+      return Stream.concat(
+              options.stream().map(ExtensionOption::getOptions).flatMap(Stream::of),
+              Stream.of(interpolator.getOptions()))
+          .toArray(org.ops4j.pax.exam.Option[]::new);
+    }
+
+    @Override
+    public String toString() {
+      return "AnnotationOptions" + options;
+    }
+  }
+
+  /**
+   * Corresponds to an extension and its annotation which allows for re-interpolation on subsequent
+   * attempts to retrieved the extension's options if an interpolation failure due to PaxExam not
```

Comment:
✏️ `attempts to retrieve the...`
✏️ `... interpolation failure occurred due to...`